source/PrintData.cpp

```diff
@@ -0,0 +1,681 @@
+/* PrintData.cpp
+Copyright (c) 2022 by warp-core
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "PrintData.h"
+
+#include "GameData.h"
+#include "Outfit.h"
+#include "Planet.h"
+#include "Ship.h"
+#include "System.h"
+
+#include <iostream>
+#include <map>
+#include <set>
+
+using namespace std;
+
+
+
+void PrintData::Print(const char *const *argv)
+{
+	const char *const *it = argv + 1;
+	if(!*it)
+		return;
+	string arg = *it;
+	if(arg == "-s" || arg == "--ships")
+		Ships(argv);
+	else if(arg == "-w" || arg == "--weapons")
+		PrintWeaponStats();
+	else if(arg == "-e" || arg == "--engines")
+		PrintEngineStats();
+	else if(arg == "--power")
+		PrintPowerStats();
+	else if(arg == "-o" || arg == "--outfits")
+		Outfits(argv);
+	else if(arg == "--planets")
+		Planets(argv);
+	else if(arg == "--systems")
+		Systems(argv);
+	cout.flush();
+}
+
+
+
+void PrintData::Help()
+{
+	cerr << "    -s, --ships: prints a table of ship stats (just the base stats, not considering any stored outfits)." << endl;
+	cerr << "    -s --sales: prints a table of ships with every 'shipyard' each appears in." << endl;
+	cerr << "    -s --loaded: prints a table of ship stats accounting for installed outfits. Does not include variants." << endl;
+	cerr << "    -s --list: prints a list of all ship names." << endl;
```

Comment:
It's not at all obvious (at least to me), that the last three arguments belong to `--ships`. Maybe format it like this? idk

```
-s, --ships: ...
    --sales: ...
    --loaded
```